WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -50,7 +50,8 @@ struct ProductSelectorView: View {
                 }
                 .buttonStyle(LinkButtonStyle())
                 .fixedSize()
-                .renderedIf(configuration.clearSelectionEnabled && viewModel.totalSelectedItemsCount > 0 && viewModel.syncStatus == .results)
+                .renderedIf(viewModel.supportsMultipleSelection)
+                .disabled(viewModel.totalSelectedItemsCount == 0 || viewModel.syncStatus != .results)
```

Comment:
Nit: we could also add a new variable to the view model and name it something like `shouldDisableClearSelectionButton`. That way we can test the logic too.

Reply:
That sounds good, I've added this to the improvements board on a separate GH issue: https://github.com/woocommerce/woocommerce-ios/issues/9353